### PR TITLE
[ART-8764] Setup advance operator release

### PR DESCRIPTION
**Pull request by ashwindasr**
Prepare to accommodate advance operator release

Alongside https://github.com/openshift-eng/aos-cd-jobs/pull/4075

### CODE REVIEW

File: doozer/doozerlib/cli/release_gen_assembly.py [hunk 1]
```diff
@@ -42,8 +42,7 @@ def releases_gen_assembly(ctx, name):
 @click.option("--custom", default=False, is_flag=True,
               help="If specified, weaker conformance criteria are applied "
                    "(e.g. a nightly is not required for every arch).")
-@click.option("--prerelease", is_flag=True,
-              help="Prepare the assembly for a prerelease operator advisory. Only valid for preview and candidate assemblies.")
+@click.option('--pre-ga-mode', help='Prepare the advisory for "prerelease" or "advance" operator release')
```
Comment (thegreyd) [inline doozer/doozerlib/cli/release_gen_assembly.py:45]: It would be nice to use `click.choices(["advance", "prerelease"])` here, as we do in other places

Comment (ashwindasr) [reply doozer/doozerlib/cli/release_gen_assembly.py]: added in https://github.com/openshift-eng/art-tools/pull/339/commits/17382dc779f01877da2a6a4424892e517a38b0c9

File: doozer/doozerlib/cli/release_gen_assembly.py [hunk 2]
```diff
@@ -42,8 +42,8 @@ def releases_gen_assembly(ctx, name):
 @click.option("--custom", default=False, is_flag=True,
               help="If specified, weaker conformance criteria are applied "
                    "(e.g. a nightly is not required for every arch).")
-@click.option("--prerelease", is_flag=True,
-              help="Prepare the assembly for a prerelease operator advisory. Only valid for preview and candidate assemblies.")
+@click.option("--pre-ga-mode", default="none", type=click.Choice(["prerelease", "advance"], case_sensitive=False),
```
Comment (thegreyd) [inline doozer/doozerlib/cli/release_gen_assembly.py:45]: Same here, we can leave the default out which will make it None when not provided

File: pyartcd/pyartcd/pipelines/gen_assembly.py [hunk 3]
```diff
@@ -251,8 +251,8 @@ async def _create_or_update_pull_request(self, assembly_definition: OrderedDict)
               help="Allow matching nightlies built from matching commits but with inconsistent RPMs")
 @click.option("--custom", is_flag=True,
               help="Custom assemblies are not for official release. They can, for example, not have all required arches for the group.")
-@click.option("--prerelease", is_flag=True,
-              help="Prepare the assembly for a prerelease operator advisory")
+@click.option("--pre-ga-mode", default="none", type=click.Choice(["prerelease", "advance"], case_sensitive=False),
```
[the remark this answers is not in the document]
Comment (ashwindasr) [reply pyartcd/pyartcd/pipelines/gen_assembly.py]: thanks, added in https://github.com/openshift-eng/art-tools/pull/339/commits/19496fcd809131a42d9794e14d0a57142ada1afd

File: doozer/doozerlib/cli/release_gen_assembly.py (hunk 2 above)
Comment (ashwindasr) [reply doozer/doozerlib/cli/release_gen_assembly.py]: added in https://github.com/openshift-eng/art-tools/commit/19496fcd809131a42d9794e14d0a57142ada1afd

**Pull request merged**